ts/src/header-validator/validate-json.ts

```diff
@@ -1335,6 +1373,78 @@ function eventTriggerData(
   )
 }
 
+function positiveUint32(j: Json, ctx: Context): Maybe<number> {
+  return number(j, ctx)
+    .filter(isInteger, ctx)
+    .filter(isInRange, ctx, 1, UINT32_MAX)
+}
+
+function maxEventStates(
+  j: Json,
+  ctx: SourceContext,
+  attributionScopeLimit: Maybe<number | null>
+): Maybe<number> {
+  return number(j, ctx)
+    .filter(isInteger, ctx)
+    .filter(isInRange, ctx, 1, ctx.vsv.maxTriggerStateCardinality)
+    .filter((n) => {
+      if (
+        attributionScopeLimit.value === null &&
+        n !== constants.defaultMaxEventStates
+      ) {
+        ctx.error('non-default if attribution_scope_limit is not set')
```

Comment:
Maybe `non-default when attribution_scope_limit is not set`?

Reply:
good point. Done.

Reply:
Please update the message to “must be default…” to avoid confusion.

Reply:
Done.